engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -320,65 +320,91 @@ class WorkflowActor(workflowToStart: WorkflowToStart,
-        jobStoreActor = jobStoreActor,
-        subWorkflowStoreActor = subWorkflowStoreActor,
-        callCacheReadActor = callCacheReadActor,
-        callCacheWriteActor = callCacheWriteActor,
-        workflowDockerLookupActor = workflowDockerLookupActor,
-        jobRestartCheckTokenDispenserActor = jobRestartCheckTokenDispenserActor,
-        jobExecutionTokenDispenserActor = jobExecutionTokenDispenserActor,
-        backendSingletonCollection,
-        initializationData,
-        startState = data.effectiveStartableState,
-        rootConfig = conf,
-        totalJobsByRootWf = totalJobsByRootWf,
-        fileHashCacheActor = fileHashCacheActorProps map context.system.actorOf,
-        blacklistCache = blacklistCache), name = s"WorkflowExecutionActor-$workflowId")
+        ioActor,
+        serviceRegistryActor,
+        restarting
+      ),
+      name)
+  }
 
+  when(InitializingWorkflowState) {
+    case Event(StartInitializing, data @ WorkflowActorData(_, Some(workflowDescriptor), _, _, _, _, _, _, _)) =>
+      val initializerActor = createInitializationActor(workflowDescriptor, s"WorkflowInitializationActor-$workflowId-${data.failedInitializationAttempts + 1}")
+      initializerActor ! StartInitializationCommand
+      goto(InitializingWorkflowState) using data.copy(currentLifecycleStateActor = Option(initializerActor), workflowDescriptor = Option(workflowDescriptor))
+    case Event(WorkflowInitializationSucceededResponse(initializationData), data @ WorkflowActorData(_, Some(workflowDescriptor), _, _, _, _, _, _, _)) =>
+      val dataWithInitializationData = data.copy(initializationData = initializationData)
+      val executionActor = createWorkflowExecutionActor(workflowDescriptor, dataWithInitializationData)
       executionActor ! ExecuteWorkflowCommand
-
       val nextState = data.effectiveStartableState match {
         case RestartableAborting => WorkflowAbortingState
         case _ => ExecutingWorkflowState
       }
-      goto(nextState) using data.copy(currentLifecycleStateActor = Option(executionActor), initializationData = initializationData)
-    case Event(WorkflowInitializationFailedResponse(reason), data @ WorkflowActorData(_, Some(workflowDescriptor), _, _, _, _, _, _)) =>
-      finalizeWorkflow(data, workflowDescriptor, Map.empty, CallOutputs.empty, Option(reason.toList))
+      goto(nextState) using dataWithInitializationData.copy(currentLifecycleStateActor = Option(executionActor))
+    case Event(WorkflowInitializationFailedResponse(reason), data @ WorkflowActorData(_, Some(workflowDescriptor), _, _, _, _, _, _, _)) =>
+      val failedInitializationAttempts = data.failedInitializationAttempts + 1
+      if (failedInitializationAttempts < maxInitializationAttempts) {
+        workflowLogger.info(s"Initialization failed on attempt ${data.failedInitializationAttempts + 1}. Will retry in 30 seconds", CromwellAggregatedException(reason, "Initialization Failure"))
```

Comment:
Probably this line should refer to the `initializationRetryInterval`.

Reply:
🤦